opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -589,20 +599,28 @@ def __init__(
             OTEL_LINK_ATTRIBUTE_COUNT_LIMIT,
             _DEFAULT_OTEL_LINK_ATTRIBUTE_COUNT_LIMIT,
         )
+
         self.max_attribute_length = self._from_env_if_absent(
             max_attribute_length,
+            OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT,
+        )
+        self.max_span_attribute_length = self._from_env_if_absent(
+            max_span_attribute_length,
             OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT,
+            # use global attribute length limit as default
+            self.max_attribute_length,
         )
 
     def __repr__(self):
-        return "{}(max_attributes={}, max_events={}, max_links={}, max_event_attributes={}, max_link_attributes={}, max_attribute_length={})".format(
+        return "{}(max_attributes={}, max_events={}, max_links={}, max_event_attributes={}, max_link_attributes={}, max_attribute_length={}, max_span_attribute_length={})".format(
```

Comment:
Nit: Can we switch the ordering of `span_attribute` and `attribute`? I'd like to see the model specific be defined first.

Reply:
wouldn't that break the API (change behavior) if all arguments are passed as positional?

Reply:
Oh do you mean if someone were to call `__repr__` directly?

Reply:
No, I mean if we swap max_span_attributes with max_attributes the signature will look like:

```python
SpanLimits(max_span_attributes, max_events, max_links, max_event_attributes, max_link_attributes, max_attributes)
```
which would be `model`, `global`, `global`, `model`, `model`, `global`. That doesn't look very nice. At least all `max_*_attributes` should be together. If we change the signature to 

```python
SpanLimits(max_span_attributes, max_event_attributes, max_link_attributes, max_attributes, max_links, max_events)
```

Then any users calling the function today as `SpanLimits(1, 2, 3, 4, 5)` will get unexpected behavior that'll be hard to detect as we changed the position of arguments.

Reply:
I think @lzchen didn't mean to change the order in `__init__` but just in the `__repr__` string.

Reply:
Oh, I see it now :)